services/comprehensive_blog_generator.py

```
@@ -0,0 +1,730 @@
+    """Generates complete blog posts using comprehensive AI approach."""
+    
+    def __init__(self):
+        self.ai_enabled = os.getenv("AI_COMPREHENSIVE_ENABLED", "true").lower() == "true"
+        self.ai_client = None
+        logger.info(f"Comprehensive blog generator initializing, AI enabled: {self.ai_enabled}")
+        
+        try:
+            self.voice_prompt = self._load_voice_prompt()
+            logger.info(f"Voice prompt loaded, length: {len(self.voice_prompt)}")
+        except Exception as e:
+            logger.error(f"Failed to load voice prompt: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)